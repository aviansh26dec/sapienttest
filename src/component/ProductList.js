import React from 'react';


import { Container, Table, InputGroup, FormControl } from 'react-bootstrap';

function App( {update, products} ) {
  return (
    <>
     <Container>
        <Table  bordered  className="mt-3">
            <thead>
                <tr>
                <th>#</th>
                <th>Title</th>
                <th>Qty.</th>
                <th>Price</th>
                </tr>
            </thead>
            <tbody>
                {products.length > 0 &&
                    products.map((item, index) => 
                        <tr key={item.id}>
                            <td>{index + 1}</td>
                            <td data-testid="list-item">{item.title}<hr/>{item.desc}</td>
                            <td>
                                <InputGroup>
                                    <InputGroup.Text className="cursor btn-danger" onClick={() => update('remove', index)}>-</InputGroup.Text>
                                        <FormControl value={(item.total === 0) ? 1 : item.total} readOnly />
                                    <InputGroup.Text className="cursor btn-success" onClick={() => update('add', index)}>+</InputGroup.Text>
                                </InputGroup>
                            </td>
                            <td>{item.currency}{item.price}</td>
                        </tr>
                    )
                }
            </tbody>
        </Table>
     </Container>
    </>
  );
}

export default App;