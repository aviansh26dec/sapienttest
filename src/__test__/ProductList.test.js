import { render, screen, wrapper } from '@testing-library/react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import ProductList from '../component/ProductList';

const product = {
    "products": [
      {
        "id": "123442",
        "title": "Product 1",
        "desc": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor",
        "image": "/product1.jpeg",
        "price": "39",
        "currency": "$"
      },
      {
        "id": "123443",
        "title": "Product 2",
        "desc": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor",
        "image": "/product2.jpeg",
        "price": "39",
        "currency": "$"
      }
    ]
  };

describe("ProductList", () => {
  it('renders without crashing', () => {
      const div = document.createElement("div");
      ReactDom.render(<ProductList products={product} />, div);
  });

  it('renders correctly', () => {
    const tree = renderer
    .create(<ProductList products={product} />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
