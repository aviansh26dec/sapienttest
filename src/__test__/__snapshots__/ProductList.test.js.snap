// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList renders correctly 1`] = `
<div
  className="container"
>
  <table
    className="mt-3 table table-bordered"
  >
    <thead>
      <tr>
        <th>
          #
        </th>
        <th>
          Title
        </th>
        <th>
          Qty.
        </th>
        <th>
          Price
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
